# 20180328 점프투파이썬 2장 마크다운은 글을 쓰는거다 제목기능
# 테스트
- 숫자형
    1.기능1
    2.기능2
- 문자열
- 리스트


In [1]:
print("3번째 수업")

3번째 수업


## 1절 숫자형

In [2]:
123

123

In [3]:
456

456

In [4]:
1+2

3

In [5]:
1*4

4

In [6]:
7/4

1.75

In [7]:
-7/4

-1.75

In [8]:
7//4

1

In [9]:
7%4

3

In [10]:
1-0.9

0.09999999999999998

0.1이아니라 구구구구구로 나온다 

In [22]:
revenue = 98456
cost = 45000
profit = revenue - cost
print(profit)

53456


## 2절 문자열

In [17]:
boy = 'I am a boy'
boy


'I am a boy'

보이 변수입력, boy를 출력표출해라

In [19]:
girl = 'I am a girl'
girl

'I am a girl'

In [20]:
friend ="""
I am a boy
I am a girl
"""
friend

'\nI am a boy\nI am a girl\n'

In [14]:
boy + ' ' + girl #빈칸 산입

"I am a boy I'm a girl"

In [20]:
boy * 3

'I am a boyI am a boyI am a boy'

In [24]:
# 문자열 인덱싱
a = " life is too short, You need Python "
a[0] + a[10]



' o'

In [25]:
# 문자열 슬라이싱 a <= : < b
a[12:17]


' shor'

In [26]:
a[:]

' life is too short, You need Python '

### Lab 1
#### 수익 계산
- 변수 revenue를 생성 후 98456 할당
- 변수 코스트 할당
- 변수 프로핏은 수익ㅇ와 비용차이값
- 프로핏 출력

### Lab 2
#### 문자열 다루기
- 변수 a를 생성해서, 문자열 아이러브 파이썬

## Lab 2 이어서
-변수 a를 생성하여 문자열
-a에서 첫번째 오의 위치를 파인드함수
-비에서 첫번째 오ㅗ의 위치를 인덱스 함수로 찾아
-a의 오 문자개수를 세라
-

In [67]:
a =  "I love Python!"
b = "You hate Python"

In [61]:
a.find('o')

3

In [62]:
a.index('o')

3

In [63]:
a.count('o')

2

In [64]:
#변수에이를 모두 대문자로 
a.upper()

'I LOVE PYTHON!'

In [65]:
#변수 비를 모두 소문자로
b.lower()

'you hate python'

In [79]:
#변수 a_6에 a의 첫7문자열  공간들로 이울어진 부분 문자열을할당하시오
a_6 = a[0:7]
print(a_6)

I love 


In [80]:
a_6.rstrip()

'I love'

In [28]:
a = "20180328Sunny"
year = a[:4]
month = a[4:6]
day = a[6:8]
weather = a[8:]
print(year, month, day, weather)

2018 03 28 Sunny


In [37]:
"{year}년 {month}월 {day}일 날씨는 {weather} 입니다." .format (year=year, month=month, day=day, weather=weather) 

'2018년 03월 28일 날씨는 Sunny 입니다.'

문자열 내장함수

In [43]:
#문자 개수찾기, 비 몇개냐
a = "hobby"
a.count('b')

2

In [44]:
#위치 알려주기, b위치
a = "hobby"
a.find('b')

2

In [45]:
#위치 알려주기 2
a.index('b')

2

In [48]:
#문자열 산입하기, abcd사이에 , 넣기
a = ", "
a.join("abcd")

'a, b, c, d'

In [49]:
#대문자로
a = "I love"
a.upper()

'I LOVE'

In [51]:
#소문자로 
a. lower

<function str.lower>

In [52]:
#왽쪽공백 지우기
a = " hi"
a.lstrip()

'hi'

In [53]:
#오른쪽공백지우기
a.rstrip()

' hi'

In [54]:
a.strip()

'hi'

In [55]:
#문자열 나누기, 문장을 단어로 나눔
a = "Life is too short"
a.split()

['Life', 'is', 'too', 'short']

### Lab 3
## 문자열 포매팅

In [40]:
pi = 3.141592
sq_root = 1.41429

#### 다음의 출력을 포매팅을 사용하여 만드시오

Pi is 3.141592, and square root of 2 is 1.41429

In [42]:
"Pi is {pi}, and square root of 2 is {sq_root}" .format (pi=pi, sq_root=sq_root)

'Pi is 3.141592, and square root of 2 is 1.41429'

파이는 파이에서, 왼쪽은은 중괄호 파이, 오른쪽파이는 위에 3.14파이말하는거임

## 3절 리스트